functions/New-DbaAgentJob.ps1

```diff
@@ -409,8 +409,7 @@ function New-DbaAgentJob {
 
             Add-TeppCacheItem -SqlInstance $server -Type job -Name $Job
 
-            # Return the job
-            $currentjob
+            Get-DbaAgentJob -SqlInstance $instance | Where-Object Name -eq $Job
```

Comment:
This is going to cause a severe performance impact because we are pulling every agent job on an instance before we filter.

It should pass the name to the job parameter so we filter the results pulling from the server.

Reply:
Good point - just overlooked that. Will open a PR for that.